pkg/app/instances/docker.go

```diff
@@ -117,16 +119,56 @@ func (m *DockerInstanceManager) DeleteHost(zone string, _ accounts.User, host st
 		return nil, fmt.Errorf("Failed to remove docker container: %w", err)
 	}
 	return &apiv1.Operation{
-		Name: host,
-		Done: true,
+		Name: "deletehost_" + host,
+		Done: false,
 	}, nil
 }
 
-func (m *DockerInstanceManager) WaitOperation(zone string, _ accounts.User, _ string) (any, error) {
+func (m *DockerInstanceManager) WaitCreateHostOperation(host string) (any, error) {
+	ctx := context.TODO()
+	for {
+		res, err := m.Client.ContainerInspect(ctx, host)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to inspect docker container: %w", err)
+		}
+		if res.State.Running {
+			return &apiv1.HostInstance{
+				Name: host,
+			}, nil
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func (m *DockerInstanceManager) WaitDeleteHostOperation(host string) (any, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Minute)
+	defer cancel()
+	resCh, errCh := m.Client.ContainerWait(ctx, host, "")
+	select {
+	case <-ctx.Done():
+		return nil, fmt.Errorf("Timeout while waiting for deleting host")
+	case err := <-errCh:
+		return nil, fmt.Errorf("Error is thrown while waiting for deleting host: %w", err)
+	case <-resCh:
+		return &apiv1.HostInstance{
+			Name: host,
+		}, nil
+	}
+}
+
+func (m *DockerInstanceManager) WaitOperation(zone string, _ accounts.User, name string) (any, error) {
 	if zone != "local" {
 		return nil, fmt.Errorf("Invalid zone. It should be 'local'.")
 	}
-	return nil, fmt.Errorf("%T#WaitOperation is not implemented", *m)
+	words := strings.SplitN(name, "_", 2)
```

Comment:
let's isolate the logic of encoding operation names and decoding operation names and add unit test to it.

we could have two functions: `EncodeOperationName(type, host string)` and `DecodeOperationName(name string) (string, string, error)`. Feel free to create a new package `docker` as these function names might be common in future instance manager implementations

this would help us be more robust and avoid crashing down below when using `words[1]` if `words` has length 1 for example.

Reply:
Introduced `EncodeOperationName` and `DecodeOperationName`, and added unit tests for them.